recipes/UnitTest_Code_Gen/unit_test_cases_gen.ipynb

```diff
@@ -0,0 +1,593 @@
+   "metadata": {},
+   "source": [
+    "## Introduction and Setup\n",
+    " \n",
+    "This recipe demonstrates how to generate unit tests for Python functions and classes using inference calls against a model hosted remotely on [Replicate](https://replicate.com/). This recipe targets developers who are looking to streamline the process of creating unit tests with minimal manual effort. The user inputs Python code and returns unit test code, incorporating \"test doubles\" for external dependencies.  The notebook depends on Granite [`Utils`](https://github.com/ibm-granite-community/utils) package for integration with LLMs using the [Langchain](https://www.langchain.com/) framework.\n",
+    "\n",
+    "### Pre-requisites\n",
+    "\n",
+    "To run this notebook, ensure you have the following:\n",
+    "\n",
+    "1. Python version: 3.9 or higher\n",
+    "2. A Replicate API token. See the `../recipes/Getting_Started_with_Granite_Code.ipynb` for details.\n",
+    "\n",
+    "### Model Details:\n",
+    "\n",
+    "1. Model Platform : Replicate\n",
+    "2. Model : IBM Granite 20b Code Instruct 8k\n",
+    "3. Model Version : ibm-granite/granite-20b-code-instruct-8k:409a0c68b74df416c7ae2a3f1552101123356f5a2c6e46d681629b62904c605b\n",
+    "\n",
+    "### Program \n",
+    "\n",
+    "1. Input: Python code/snippets with instructions for test packages that need to utilized and optional type of unit test case scenarios to be covered.\n",
+    "2. Output: Python code with unit test packages and libraries, test doubles, assert implementation for Unit testing of given input.\n",
+    "\n",
+    "### Disclaimer\n",
+    "\n",
+    "Results using the 20b code instruct Granite model, with an 8k context, are generally better than the outputs when using the 8b code instruct Granite model, with an 128k context. In either case, the code generated may require additional modifications, depending on the test libraries requested and other aspects of the user input. "
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "8caa29bb",
+   "metadata": {},
+   "source": [
+    "### Install required Langchain and replicate packages"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 1,
+   "id": "79545db2",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "!pip install langchain_community replicate"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "b9c45de3",
+   "metadata": {},
+   "source": [
+    "### Install Granite `utils` package\n",
+    "\n",
+    "This package is a thin shim with various functions that are required for notebooks.\n",
+    "\n",
+    "To see the implementation of its functions, see the [utils repo](https://github.com/ibm-granite-community/utils/tree/main)."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "54f7fa00",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "!pip install git+https://github.com/ibm-granite-community/utils"
```

Comment:
This and the `pip install langchain_community replicate` above can be pulled in with:
`!pip install git+https://github.com/ibm-granite-community/granite-kitchen`

As shown [here](https://github.com/ibm-granite-community/granite-kitchen/blob/main/recipes/Getting_Started/Getting_Started_with_Replicate.ipynb). You may just want to check that you are getting the same version of replicate.